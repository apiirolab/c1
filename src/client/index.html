<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>