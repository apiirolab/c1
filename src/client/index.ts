export const info = 'Client Info';
