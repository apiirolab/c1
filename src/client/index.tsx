import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
