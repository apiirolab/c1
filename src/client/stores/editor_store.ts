import { types } from 'mobx-state-tree';

export const EditorStore = types
  .model('EditorStore', {
    dirty: types.maybe(types.boolean)
  })
  .actions(self => ({
    setDirty(value: boolean) {
      self.dirty = value;
    }
  }));

export const editorStore = EditorStore.create({});
