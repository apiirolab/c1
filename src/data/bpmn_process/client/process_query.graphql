query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
