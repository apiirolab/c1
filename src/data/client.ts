import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
