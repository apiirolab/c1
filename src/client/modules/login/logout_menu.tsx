import * as React from 'react';

import { gql, QueryTypes } from 'data/client';

import { inject, observer } from 'mobx-react';
import { Query } from 'react-apollo';
import { Divider, Dropdown, Icon, Loader, Menu } from 'semantic-ui-react';

import { LocalStorage } from '../../config/local_storage';

import * as RESUME_NF from './queries/resume_query.graphql';
import * as USER_FRAGMENT from './queries/user.fragment.graphql';

export const RESUME = gql`
  ${RESUME_NF}
  ${USER_FRAGMENT}
`;

/* ========================================================= 
    QUERY
   ======================================================== */

export class ResumeQuery extends Query<QueryTypes.Resume, QueryTypes.ResumeVariables> {
  static displayName = 'ResumeQuery';
}

/* =========================================================
    COMPONENT
   ======================================================== */

type Props = {
  store?: App.Store;
};

@inject('store')
@observer
export class LogoutMenu extends React.Component<Props> {
  logout = () => {
    this.props.store.logout();
  };

  onCompleted = (data: QueryTypes.Resume | {}) => {
    this.props.store.setUser((data as QueryTypes.Resume).resume.user);
  };

  onError = () => {
    if (process.env.NODE_ENV !== 'development') {
      this.props.store.logout();
    }
  };

  render() {
    if (!this.props.store.userId) {
      return null;
    }
    return (
      <ResumeQuery
        query={RESUME}
        variables={{ token: LocalStorage.token }}
        onCompleted={this.onCompleted}
        onError={this.onError}
      >
        {({ loading, error, data }) => {
          /* == ERROR == */
          if (error) {
            LocalStorage.userId = null;
            LocalStorage.token = null;
            return null;
          }
          /* == LOADING == */
          if (loading || !data) {
            return (
              <Menu.Item>
                <Loader active inline size="mini" />
              </Menu.Item>
            );
          }

          if (!data.resume.token) {
            LocalStorage.userId = null;
            LocalStorage.token = null;
            return null;
          }

          return (
            <Dropdown
              item
              trigger={
                <span>
                  <Icon name="user" />
                  {data.resume.user.name}
                </span>
              }
            >
              <Dropdown.Menu>
                {this.props.children}
                <Divider />
                <Dropdown.Item
                  icon="sign out"
                  onClick={this.logout}
                  content={this.props.store.i18n`Logout`}
                />
              </Dropdown.Menu>
            </Dropdown>
          );
        }}
      </ResumeQuery>
    );
  }
}
