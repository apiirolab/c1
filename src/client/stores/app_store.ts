import i18n from 'es2015-i18n-tag';
import { types } from 'mobx-state-tree';

import { client } from 'client/config/apollo';
import { QueryTypes } from 'data/client';
import { LocalStorage } from '../config/local_storage';
import { LoginStore } from '../modules/login/login_store';
import { UserStore } from '../modules/user/user_store';

declare global {
  namespace App { type Store = typeof AppStore.Type; }
}

export const AppStore = types
  .model('AppStore', {
    userId: types.maybe(types.string),
    user: types.maybe(UserStore),
    login: types.optional(LoginStore, () => LoginStore.create({}))
  })
  .volatile(_self => ({
    i18n,
    client,
    localStorage: LocalStorage
  }))
  .actions(self => {
    return {
      setUser(user: QueryTypes.User) {
        self.login.password = '';
        self.userId = user.id;
        self.user = user as any;
        self.localStorage.userId = user.id;
      },
      logout() {
        self.userId = undefined;
        self.user = undefined;
        self.client().resetStore();
        self.localStorage.userId = null;
        self.localStorage.token = null;
      }
    };
  });

export const appStore = AppStore.create({ userId: LocalStorage.userId });
