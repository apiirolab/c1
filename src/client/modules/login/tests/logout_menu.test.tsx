import * as React from 'react';

import 'jest-styled-components';

import { AppStore } from 'client/stores/app_store';
import { create, mock, MockedProvider } from 'tests/client';

import RESUME_MUTATION = require('data/client/resume_query.graphql');
import { Dropdown, Menu } from 'semantic-ui-react';
import { LogoutMenu, ResumeQuery } from '../logout_menu';

describe('Logout', function() {
  describe('Menu Button', () => {
    const CreateLogoutMenu = (token: string, store = AppStore.create()) =>
      create(
        <MockedProvider>
          <Menu>
            <LogoutMenu store={store} token={token} />
          </Menu>
        </MockedProvider>
      );

    beforeEach(() => {
      mock.reset();
    });

    it('renders empty if no user present and loading', () => {
      mock.expect(RESUME_MUTATION).loading();

      const component = CreateLogoutMenu('1234');

      expect(component).toMatchSnapshot();
      component.unmount();
    });

    it('renders empty if no user present', () => {
      mock.expect(RESUME_MUTATION).reply({});

      const component = CreateLogoutMenu('1234');

      expect(component).toMatchSnapshot();
      component.unmount();
    });

    it('renders loading ', () => {
      mock.expect(RESUME_MUTATION).loading();

      const store = AppStore.create({ userId: '1' });
      const component = CreateLogoutMenu('1234', store);
      expect(component).toMatchSnapshot();

      component.unmount();
    });

    it('renders empty if no user returned ', () => {
      mock.expect(RESUME_MUTATION).reply({ resume: {} });

      const store = AppStore.create({ userId: '1' });
      const component = CreateLogoutMenu('1234', store);
      expect(component).toMatchSnapshot();

      component.unmount();
    });

    it('renders fail and resets store', () => {
      mock.expect(RESUME_MUTATION).fail('Token Fail');

      const store = AppStore.create({ userId: '1' });
      const component = CreateLogoutMenu('1234', store);
      expect(component).toMatchSnapshot();

      // test callback

      const query = component.root.findByType(ResumeQuery);
      query.props.onError();
      expect(store.userId).not.toBeDefined();

      component.unmount();
    });

    it('renders and sets value', () => {
      mock
        .expect(RESUME_MUTATION)
        .reply({ resume: { user: { id: '2', name: 'Tomas Trescak' }, token: 'ABCD' } });

      // const d = component.root.findByType(LogoutMenu).instance;
      // const spy = jest.fn();
      // d.wrappedInstance.onError = spy;

      const store = AppStore.create({ userId: '1' });
      const component = CreateLogoutMenu('1234', store);
      expect(component).toMatchSnapshot();

      // test callback

      const query = component.root.findByType(ResumeQuery);
      const user = { id: '2', name: 'Tomi' };
      query.props.onCompleted({ resume: { user } });
      expect(store.userId).toBe('2');
      expect(store.user).toEqual(user);

      component.unmount();
    });

    it('logs out', () => {
      mock
        .expect(RESUME_MUTATION)
        .reply({ resume: { user: { id: '2', name: 'Tomas Trescak' }, token: 'ABCD' } });

      const store = AppStore.create({ userId: '1', user: { id: '1', name: 'Tomas' } });
      const component = CreateLogoutMenu('1234', store);

      component.root.findByProps({ icon: 'sign out' }).props.onClick();

      expect(store.userId).not.toBeDefined();
      expect(store.user).not.toBeDefined();
    });

    return {
      component: CreateLogoutMenu('123')
    };
  });
});
