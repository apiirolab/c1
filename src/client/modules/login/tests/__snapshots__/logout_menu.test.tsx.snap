// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Menu Button renders and sets value 1`] = `
<style>
</style>

<div
  className="ui menu"
>
  <div
    aria-expanded={false}
    className="ui item dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <span>
      <i
        aria-hidden="true"
        className="user icon"
      />
      Tomas Trescak
    </span>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        className="ui divider"
      />
      <div
        className="item"
        onClick={[Function]}
        role="option"
      >
        <i
          aria-hidden="true"
          className="sign out icon"
        />
        <span
          className="text"
        >
          Logout
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Logout Menu Button renders empty if no user present 1`] = `
<style>
</style>

<div
  className="ui menu"
/>
`;

exports[`Logout Menu Button renders empty if no user present and loading 1`] = `
<style>
</style>

<div
  className="ui menu"
/>
`;

exports[`Logout Menu Button renders empty if no user returned  1`] = `
<style>
</style>

<div
  className="ui menu"
/>
`;

exports[`Logout Menu Button renders fail and resets store 1`] = `
<style>
</style>

<div
  className="ui menu"
/>
`;

exports[`Logout Menu Button renders loading  1`] = `
<style>
</style>

<div
  className="ui menu"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="ui mini active inline loader"
    />
  </div>
</div>
`;
